pkg/kubecost/summaryallocation_json.go

```diff
@@ -0,0 +1,116 @@
+package kubecost
+
+import (
+	"math"
+	"time"
+)
+
+// SummaryAllocationResponse is a sanitized version of SummaryAllocation, which
+// formats fields and protects against issues like mashaling NaNs.
+type SummaryAllocationResponse struct {
+	Name                   string    `json:"name"`
```

Comment:
In addition, it seems like Go has native JSON encoding/decoding capabilities. If so, that's super neat!

Reply:
Yep, you've got the idea exactly. Go does indeed support JSON natively, even if you don't specify `json:"name"` (this is called a "struct tag"). It's worth noting that `encoding/json` is only able to marshal/unmarshal into exported (capitalized) fields by default. Otherwise you need to write a custom marshal/unmarshal function.

Proper docs here are pretty good: https://pkg.go.dev/encoding/json#Marshal

Reply:
Very cool! I'll definitely take a look through those docs.